R/apa_print_glht.R

```diff
@@ -58,237 +58,43 @@ apa_print.summary.glht <- function(
 
   tidy_x <- broom::tidy(x)
   test_stat <- ifelse(x$df == 0, "z", paste0("t(", x$df, ")"))
-  conf_level <- paste0(ci * 100, "\\% CI")
+  conf_level <- paste0(ci * 100, "% CI")
+  p_value <- names(tidy_x)[grepl("p.value", names(tidy_x), fixed = TRUE)]
 
   # Assamble table
   ## Add (adjusted) confidence intervall
   multcomp_adjustment <- if(x$test$type == "none") multcomp::univariate_calpha() else multcomp::adjusted_calpha()
   print_ci <- stats::confint(x, level = ci, calpha = multcomp_adjustment)$confint
   dimnames(print_ci) <- NULL
   table_ci <- unlist(print_confint(print_ci[, -1], ...)) # Remove point estimate from matrix
-  contrast_table <- cbind(estimate = tidy_x$estimate, confint = table_ci, tidy_x[, c("statistic", "p.value")])
-  rownames(contrast_table) <- sanitize_terms(tidy_x$lhs)
-
-  ## Format numbers
-  contrast_table$estimate <- printnum(contrast_table$estimate, ...)
-  contrast_table$statistic <- printnum(contrast_table$statistic, digits = 2)
-  contrast_table$p.value <- printp(contrast_table$p.value)
-
-  # Concatenate character strings and return as named list
-  apa_res <- init_apa_results()
-
-  apa_res$estimate <- apply(contrast_table, 1, function(y) {
-    paste0("$\\Delta M = ", y["estimate"], "$, ", conf_level, " ", y["confint"])
-  })
-
-  apa_res$statistic <- apply(contrast_table, 1, function(y) {
-    paste0("$", test_stat, " = ", y["statistic"], "$, $p ", add_equals(y["p.value"]), "$")
-  })
-
-  apa_res$full_result <- paste(apa_res$estimate, apa_res$stat, sep = ", ")
-  names(apa_res$full_result) <- names(apa_res$estimate)
-
-  apa_res <- lapply(apa_res, as.list)
-
-  # Add table
-  rownames(contrast_table) <- tidy_x$lhs
-  colnames(contrast_table) <- c("estimate", "ci", "statistic", "p.value")
-  variable_label(contrast_table) <- c(
-    estimate = "$\\Delta M$"
-    , ci = conf_level
+  tidy_x$std.error <- table_ci
+  colnames(tidy_x)[colnames(tidy_x) == "std.error"] <- "conf.int"
+
+  ## Typeset columns
+  tidy_x$contrast <- sanitize_terms(tidy_x$contrast)
+  tidy_x$estimate <- printnum(tidy_x$estimate, ...)
+  tidy_x$statistic <- printnum(tidy_x$statistic, digits = 2)
+  tidy_x[[p_value]] <- printp(tidy_x[[p_value]])
+
+  ## Add variable labels
+  variable_labels(tidy_x) <- c(
+    contrast = "Contrast"
+    , null.value = "$\\mu_0$"
+    , estimate = "$\\Delta M$"
+    , conf.int = conf_level
     , statistic = paste0("$", test_stat, "$")
-    , p.value = "$p$"
   )
-  apa_res$table <- contrast_table
-  attr(apa_res$table, "class") <- c("apa_results_table", "data.frame")
-  apa_res
-}
-
-#' @rdname apa_print.glht
-#' @export
-
-# apa_print.emmGrid <- function(x, ...) {
-#   ellipsis <- list(...)
-#   if(is.null(ellipsis$infer)) ellipsis$infer <- TRUE
-#   ellipsis$object <- x
-#
-#   summary_x <- do.call("summary", ellipsis)
-#   apa_print(summary_x, ...)
-# }
+  variable_labels(tidy_x[[p_value]]) <- if(p_value == "p.value") "$p$" else if(p_value == "adj.p.value") "$p_{adj}$"
```

Comment:
Hmm, what would be a scenario where we don't have a p-value column?

Reply:
Ok, this output comes from broom, and it should therefore only be `p.value` or `adj.p.value`, but we have a large variety of different names for this column in our lookup tables (`Pr(>|t|)`, etc). While I hope that the broom maintainers will not introduce anything new (consider bootstrapped *p* values), I'm getting stomach aches thinking that our code only works if the exact column names are present in the digested object. Again, this whole bug was introduced because the name of this column changed in broom.